app/src/pages/playground/PlaygroundResponseFormat.tsx

```diff
@@ -59,29 +63,45 @@ export function PlaygroundResponseFormat({
       p.invocationName === RESPONSE_FORMAT_PARAM_NAME ||
       p.canonicalName === RESPONSE_FORMAT_PARAM_CANONICAL_NAME
   );
-
-  const [initialResponseFormatDefinition] = useState(
-    JSON.stringify(responseFormat?.valueJson ?? {}, null, 2)
-  );
-
+  const [initialResponseFormatDefinition, setInitialResponseFormatDefinition] =
+    useState(() => JSON.stringify(responseFormat?.valueJson ?? {}, null, 2));
   const currentValueRef = useRef(initialResponseFormatDefinition);
+  const store = usePlaygroundStore();
+
+  // when the instance provider changes, we need to update the editor value
+  // to reflect the new response format schema
+  useEffect(() => {
+    const state = store.getState();
+    const instance = state.instances.find((i) => i.id === playgroundInstanceId);
+    if (instance == null) {
+      return;
+    }
+    const responseFormat = instance.model.invocationParameters.find(
+      (p) =>
+        p.invocationName === RESPONSE_FORMAT_PARAM_NAME ||
+        p.canonicalName === RESPONSE_FORMAT_PARAM_CANONICAL_NAME
+    );
+    if (responseFormat == null) {
+      return;
+    }
+    const newResponseFormatDefinition = JSON.stringify(
+      responseFormat.valueJson,
+      null,
+      2
+    );
+    if (
+      newResponseFormatDefinition != null &&
+      newResponseFormatDefinition !== "null"
```

Comment:
this is sorta repeated logic but might not be intuitive as to why you are checking a string null. I think it's probably best to move this into a util and describe why you have to check both. That way this is easier to read

Reply:
good call, will extend isJSONString with null exclusion

Reply:
Fixed